src/wp-admin/includes/image.php

```diff
@@ -393,79 +439,114 @@ function _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id ) {
-			 * To change the behavior, unset changed/mismatched sizes in the `sizes` array in image meta.
-			 */
-			if ( array_key_exists( $size_name, $new_sizes ) ) {
-				unset( $new_sizes[ $size_name ] );
-			}
-		}
-	} else {
-		$image_meta['sizes'] = array();
+	// Set up a top level sources array for the full size images.
+	if ( ! isset( $metadata['sources'] ) || ! is_array( $metadata['sources'] ) ) {
+		$metadata['sources'] = array();
 	}
+	$mime_type = get_post_mime_type( $attachment_id );
+	$editor    = wp_get_image_editor( $file );
 
-	if ( empty( $new_sizes ) ) {
-		// Nothing to do...
+	if ( is_wp_error( $editor ) ) {
+		// The image cannot be edited.
 		return $image_meta;
 	}
 
-	/*
-	 * Sort the image sub-sizes in order of priority when creating them.
-	 * This ensures there is an appropriate sub-size the user can access immediately
-	 * even when there was an error and not all sub-sizes were created.
-	 */
-	$priority = array(
-		'medium'       => null,
-		'large'        => null,
-		'thumbnail'    => null,
-		'medium_large' => null,
-	);
+	// Determine which mime types to output.
+	$valid_mime_transforms = wp_upload_image_mime_transforms( $attachment_id );
+	$output_mime_types     = isset( $valid_mime_transforms[ $mime_type ] ) ? $valid_mime_transforms[ $mime_type ] : array( $mime_type );
+	$dirname               = pathinfo( $file, PATHINFO_DIRNAME );
 
-	$new_sizes = array_filter( array_merge( $priority, $new_sizes ) );
+	// Generate all off of the output mime types.
+	foreach ( $output_mime_types as $mime_index => $output_mime_type ) {
 
-	$editor = wp_get_image_editor( $file );
+		// Check if any of the new sizes already exist for this mime type.
+		if ( isset( $image_meta['sizes'] ) && is_array( $image_meta['sizes'] ) ) {
+			foreach ( $image_meta['sizes'] as $size_name => $size_meta ) {
+				if ( ! isset( $size_meta['sources'] ) ) {
+					$size_meta['sources'] = array();
+				}
 
-	if ( is_wp_error( $editor ) ) {
-		// The image cannot be edited.
-		return $image_meta;
-	}
+				if ( ! isset( $new_sizes['sources'][ $output_mime_type ] ) ) {
+					$new_sizes['sources'][ $output_mime_type ] = array();
+					continue;
+				}
 
-	// If stored EXIF data exists, rotate the source image before creating sub-sizes.
-	if ( ! empty( $image_meta['image_meta'] ) ) {
-		$rotated = $editor->maybe_exif_rotate();
+				/*
+				* Only checks "size name" so we don't override existing images even if the dimensions
+				* don't match the currently defined size with the same name.
+				* To change the behavior, unset changed/mismatched sizes in the `sizes` array in image meta.
+				*/
+				if ( array_key_exists( $size_name, $new_sizes['sources'][ $output_mime_type ] ) ) {
+					unset( $new_sizes[ $size_name ] );
+				}
+			}
+		} else {
+			$image_meta['sizes'] = array();
+		}
 
-		if ( is_wp_error( $rotated ) ) {
-			// TODO: Log errors.
+		if ( empty( $new_sizes ) ) {
+			// Nothing to do...
+			return $image_meta;
```

Comment:
We cannot `return` here since this is now in a `foreach` loop, otherwise the following MIME types would be entirely skipped.
```suggestion
			continue;
```

Reply:
this bit is removed in the latest patch